corehq/apps/reports/standard/deployments.py

```diff
@@ -318,22 +324,33 @@ def process_rows(self, users, fmt_for_export=False):
                 last_builds = reporting_metadata.get('last_builds')
                 if last_builds:
                     last_build = self.get_data_for_app(last_builds, self.selected_app_id)
+                if device:
+                    device_app_metas = device.get('app_meta')
+                    if device_app_metas:
+                        device_app_meta = self.get_data_for_app(device_app_metas, self.selected_app_id)
             else:
                 last_sub = reporting_metadata.get('last_submission_for_user', {})
                 last_sync = reporting_metadata.get('last_sync_for_user', {})
                 last_build = reporting_metadata.get('last_build_for_user', {})
+                app_id = None
+                if last_build:
+                    app_id = last_build.get('app_id')
+                if device is not None and app_id is not None:
+                    device_app_metas = device.get('app_meta')
+                    if device_app_metas:
+                        device_app_meta = self.get_data_for_app(device_app_metas, app_id)
```

Comment:
in case a user has not selected an app id, should we be returning the total number of unsent forms on the device?

Reply:
lets also wait for what @dimagi/product say about this.